### PR TITLE
Fixes #12735 - Fix proxy config after reset

### PR DESCRIPTION
**Pull request by johnpmitsch**

### CODE REVIEW

**Comment by parthaa** [comment]
Extra Credit: 
can you make these config entries a separate task and have setup run it.

File: lib/katello/tasks/reset.rake [hunk 1]
```diff
@@ -43,5 +43,8 @@ namespace :katello do
     # Otherwise migration and seeds fail because they require a reloaded environment
     system('rake db:migrate')
     system('rake db:seed')
+    system('rake -- config -k ssl_ca_file -v /etc/pki/katello/certs/katello-default-ca.crt')
+    system('rake -- config -k ssl_certificate -v /etc/pki/katello/certs/$(hostname)-foreman-client.crt')
+    system('rake -- config -k ssl_priv_key -v /etc/pki/katello/private/$(hostname)-foreman-client.key')
```
Comment (jlsherrill) [inline lib/katello/tasks/reset.rake]: why not go ahead and make the new smart proxy object too?

Comment (johnpmitsch) [inline lib/katello/tasks/reset.rake]: +1, can add

**Comment by johnpmitsch** [comment]
updated

File: lib/katello/tasks/reset.rake [hunk 2]
```diff
@@ -26,6 +26,14 @@ namespace :katello do
     end
   end
 
+  task :reset_default_smart_proxy do
+    hostname = Socket.gethostname.chomp
+    system("rake -- config -k ssl_ca_file -v /etc/pki/katello/certs/katello-default-ca.crt")
+    system("rake -- config -k ssl_certificate -v /etc/pki/katello/certs/#{hostname}-foreman-client.crt")
+    system("rake -- config -k ssl_priv_key -v /etc/pki/katello/private/#{hostname}-foreman-client.key")
```
Comment (jlsherrill) [inline lib/katello/tasks/reset.rake]: Why not just set the settings directly? 
```
 Setting['ssl_ca_file'] = "/etc/pki/katello/certs/katello-default-ca.crt" 
```
for example

File: lib/katello/tasks/reset.rake [hunk 3]
```diff
@@ -26,6 +26,14 @@ namespace :katello do
...
+    SmartProxy.where(name: hostname, url: "https://#{hostname}:9090").first_or_create
```
Comment (jlsherrill) [inline lib/katello/tasks/reset.rake]: This isn't working for me for some reason, You might use .first_or_create!   to see if you can get it to tell you an error (for some reason i couldn't)

Comment (jlsherrill) [inline lib/katello/tasks/reset.rake]: When i try it in the console i get:

```
Hirb Error: uninitialized constant ProxyAPI::Pulp
    /home/vagrant/git/katello/app/models/katello/concerns/smart_proxy_extensions.rb:71:in `update_puppet_path'
    /home/vagrant/git/katello/app/models/katello/concerns/smart_proxy_extensions.rb:66:in `puppet_path'
```

**Comment by johnpmitsch** [comment]
@jlsherrill updated

File: lib/katello/tasks/reset.rake [hunk 4]
```diff
@@ -26,6 +26,14 @@ namespace :katello do
     end
   end
 
+  task :reset_default_smart_proxy do
+    hostname = Socket.gethostname.chomp
+    Setting['ssl_ca_file'] = "/etc/pki/katello/certs/katello-default-ca.crt"
+    Setting['ssl_certificate'] = "/etc/pki/katello/certs/#{hostname}-foreman-client.crt"
+    Setting['ssl_priv_key'] = "/etc/pki/katello/private/#{hostname}-foreman-client.key"
+    SmartProxy.where(name: hostname, url: "https://#{hostname}:9090").first_or_create!
```
Comment (jlsherrill) [inline lib/katello/tasks/reset.rake]: can you set User.current = User.anonymous_admin here?

**Comment by jlsherrill** [comment]
APJ

**Pull request merged**